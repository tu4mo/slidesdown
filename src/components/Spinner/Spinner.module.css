@keyframes spinner {
  0% {
    transform: rotate(0);
  }
  100% {
    transform: rotate(180deg);
  }
}

.spinner {
  align-items: center;
  display: flex;
  height: 100%;
  left: 0;
  justify-content: center;
  position: absolute;
  top: 0;
  width: 100%;
}

.spinner::after {
  animation: spinner 1s infinite;
  animation-fill-mode: forwards;
  border: 2px solid var(--color-purple);
  border-radius: 4px;
  content: '';
  height: 2rem;
  width: 2rem;
}
