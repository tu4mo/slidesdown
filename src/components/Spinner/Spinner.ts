import styled from 'styled-components'

const StyledSpinner = styled.div`
  align-items: center;
  display: flex;
  height: 100%;
  left: 0;
  justify-content: center;
  position: absolute;
  top: 0;
  width: 100%;

  &::after {
    animation: spinner-animation 1s infinite;
    animation-fill-mode: forwards;
    border: 2px solid ${props => props.theme.colors.purple};
    border-radius: 4px;
    content: '';
    height: 2rem;
    width: 2rem;
  }

  @keyframes spinner-animation {
    0% {
      transform: rotate(0);
    }
    100% {
      transform: rotate(180deg);
    }
  }
`

export default StyledSpinner
