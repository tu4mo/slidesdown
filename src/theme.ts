export default {
  breakpoints: {
    md: '576px',
    xl: '768px'
  },
  colors: {
    background: '#edf1f5',
    border: '#e4e4e8',
    dimText: '#a4a4a8',
    lightText: '#646468',
    magenta: '#c86dd7',
    purple: '#3023ae'
  },
  fonts: {
    default: 'Rubik',
    monospace: 'Inconsolata'
  }
}
