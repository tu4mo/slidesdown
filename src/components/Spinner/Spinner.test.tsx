import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { ThemeProvider } from 'styled-components'

import theme from '../../theme'
import Spinner from './Spinner'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Spinner />
      </ThemeProvider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
