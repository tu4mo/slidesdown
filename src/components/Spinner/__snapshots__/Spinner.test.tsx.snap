// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  left: 0;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: absolute;
  top: 0;
  width: 100%;
}

.c0::after {
  -webkit-animation: spinner-animation 1s infinite;
  animation: spinner-animation 1s infinite;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  border: 2px solid #3023ae;
  border-radius: 4px;
  content: '';
  height: 2rem;
  width: 2rem;
}

<div
  className="c0"
/>
`;
